src/S3/StreamWrapper.php

```diff
@@ -323,10 +328,18 @@ private function createStat($path, $flags)
     private function statDirectory($parts, $path, $flags)
     {
         // Stat "directories": buckets, or "s3://"
-        if (!$parts['Bucket'] ||
-            $this->getClient()->doesBucketExist($parts['Bucket'])
-        ) {
-            return $this->formatUrlStat($path);
+        if (self::$useV2Existence) {
```

Comment:
you can restructure this if you want.  
```
if (!parts['Bucket']) {
     return $this->formatUrlStat($path); 
}   else if (self::$useV2Existence
      && $this->getClient()->doesBucketExist($parts['Bucket']
      ) {
     return $this->formatUrlStat($path); 
}  else if (!self::$useV2Existence
      && $this->getClient()->doesBucketExist($parts['Bucket']
     )  {
     return $this->formatUrlStat($path); 
      } 
}

```
Not saying this is better or worse, just offering another option where we don't duplicate the bucket check 
You could also extract calling the legacy or V2 method into another method:

```
private function checkBucketExistance($bucket) {
if (self::useV2) {
      return $this->getClient()->doesBucketExistV2($bucket)
      }
return $this->getClient()->doesBucketExist($bucket)
}
```